repos/system_upgrade/el7toel8/actors/rpmtransactionconfigtaskscollector/libraries/scanner.py

```diff
@@ -19,7 +20,11 @@ def load_tasks_file(path, logger):
 
 def load_tasks(base_dir, logger):
     # Loads configuration files to_install, to_keep, and to_remove from the given base directory
+    to_install = load_tasks_file(os.path.join(base_dir, 'to_install'), logger)
+    # we do not want to put into rpm transaction what is already installed (it will go to "to_upgrade" bucket)
+    to_install_filtered = [pkg for pkg in to_install if pkg not in api.consume(InstalledRedHatSignedRPM)]
```

Comment:
It would be worth printing a debug msg when not adding a pkg from the to_install file to the dnf plugin to_install list. Because in the end a pkg mentioned in the file may get even removed (e.g. `finger` that is no longer available on RHEL8) and it would be hard to find why the package that was to be installed got removed.

Reply:
Hm, I installed finger, ran the upgrade and got `Package finger-0.17-52.el7.x86_64 is already installed.` in the transaction, even though it should have gotten removed. I'll test it more tomorrow.

Reply:
I iterated through model objects instead of names. Fixed.